src/main/java/io/confluent/connect/jdbc/JdbcSourceConnector.java

```diff
@@ -115,25 +115,30 @@ public void start(Map<String, String> properties) throws ConnectException {
   @Override
   public List<Map<String, String>> taskConfigs(int maxTasks) {
     String query = config.getString(JdbcSourceConnectorConfig.QUERY_CONFIG);
+    List<Map<String, String>> taskConfigs;
     if (!query.isEmpty()) {
-      List<Map<String, String>> taskConfigs = new ArrayList<>(1);
+      taskConfigs = new ArrayList<>(1);
       Map<String, String> taskProps = new HashMap<>(configProperties);
       taskProps.put(JdbcSourceTaskConfig.TABLES_CONFIG, "");
       taskConfigs.add(taskProps);
-      return taskConfigs;
     } else {
       List<String> currentTables = tableMonitorThread.tables();
-      int numGroups = Math.min(currentTables.size(), maxTasks);
-      List<List<String>> tablesGrouped = ConnectorUtils.groupPartitions(currentTables, numGroups);
-      List<Map<String, String>> taskConfigs = new ArrayList<>(tablesGrouped.size());
-      for (List<String> taskTables : tablesGrouped) {
-        Map<String, String> taskProps = new HashMap<>(configProperties);
-        taskProps.put(JdbcSourceTaskConfig.TABLES_CONFIG,
-                      StringUtils.join(taskTables, ","));
-        taskConfigs.add(taskProps);
+      if (currentTables.isEmpty()) {
+        taskConfigs = new ArrayList<>();
+        log.warn("No tasks are running because no tables were found");
```

Comment:
I'd be careful about the tense of this log message. This method returns configs that are used to generate tasks, so I think future tense might be more appropriate. Something like "No tasks will be run..." instead of "No tasks are running..." but feel free to tweak if you'd like.